base/strings/string.jl

```diff
@@ -284,7 +284,7 @@ getindex(s::String, r::AbstractUnitRange{<:Integer}) = s[Int(first(r)):Int(last(
     return ss
 end
 
-length(s::String) = length_continued(s, 1, ncodeunits(s), ncodeunits(s))
+@assume_effects :foldable :removable length(s::String) = length_continued(s, 1, ncodeunits(s), ncodeunits(s))
```

Comment:
Same question here.

Reply:
Here we definitely can't because `length_continued` uses `@inbounds` and indices provided to the function making it non consistent.